auth-utility_test.go

```diff
@@ -0,0 +1,178 @@
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/goadesign/goa"
+	"github.com/goadesign/goa/middleware"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	// the various HTTP endpoints
+	endpointWorkItems              = "/api/workitems"
+	endpointWorkItemTypes          = "/api/workitemtypes"
+	endpointWorkItemLinkCategories = "/api/workitemlinkcategories"
+	endpointWorkItemLinkTypes      = "/api/workitemlinktypes"
+	endpointWorkItemLinks          = "/api/workitemlinks"
+)
+
+// Expected strcture of 401 error response
+type errorResponseStruct struct {
+	Id     string
+	Code   string
+	Status int
+	Detail string
+}
+
+// testSecureAPI defines how a Test object is.
+type testSecureAPI struct {
+	method             string
+	url                string
+	expectedStatusCode int    // this will be tested against responseRecorder.Code
+	expectedErrorCode  string // this will be tested only if response body gets unmarshelled into errorResponseStruct
+	payload            *bytes.Buffer
+	jwtToken           string
+}
+
+// getExpiredAuthHeader returns a JWT bearer token with an expiration date that lies in the past
+func getExpiredAuthHeader(t *testing.T, key interface{}) string {
+	token := jwt.New(jwt.SigningMethodRS256)
+	token.Claims = jwt.MapClaims{"exp": float64(time.Now().Unix() - 100)}
+	tokenStr, err := token.SignedString(key)
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+	return fmt.Sprintf("Bearer %s", tokenStr)
+}
+
+// getMalformedAuthHeader returns a JWT bearer token with the wrong prefix of "Malformed Bearer" instead of just "Bearer"
+func getMalformedAuthHeader(t *testing.T, key interface{}) string {
+	token := jwt.New(jwt.SigningMethodRS256)
+	tokenStr, err := token.SignedString(key)
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+	return fmt.Sprintf("Malformed Bearer %s", tokenStr)
+}
+
+// getExpiredAuthHeader returns a valid JWT bearer token
+func getValidAuthHeader(t *testing.T, key interface{}) string {
+	token := jwt.New(jwt.SigningMethodRS256)
+	tokenStr, err := token.SignedString(key)
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+	return fmt.Sprintf("Bearer %s", tokenStr)
+}
+
+// UnauthorizeCreateUpdateDeleteTest will check authorized access to Create/Update/Delete APIs
+func UnauthorizeCreateUpdateDeleteTest(t *testing.T, getDataFunc func(t *testing.T) []testSecureAPI, createServiceFunc func() *goa.Service, mountCtrlFunc func(service *goa.Service) error) {
+	resource.Require(t, resource.Database)
+
+	// This will be modified after merge PR for "Viper Environment configurations"
+	publickey, err := jwt.ParseRSAPublicKeyFromPEM((configuration.GetTokenPublicKey()))
+	if err != nil {
+		t.Fatal("Could not parse Key ", err)
+	}
+	tokenTests := getDataFunc(t)
+
+	for _, testObject := range tokenTests {
+		// Build a request
+		var req *http.Request
+		var err error
+		if testObject.payload == nil {
```

Comment:
I appreciate that, still would be nice to reduce the bogon flux here: it's obviously bogus code.